File: full/corelocation/tests/CoreLocationMixedConsumer.m
#import <CoreLocation/CoreLocation.h>

double OpenCoreLocationMixedDistance(void) {
    CLLocationCoordinate2D source = CLLocationCoordinate2DMake(40.7128, -74.0060);
    CLLocationCoordinate2D destination = CLLocationCoordinate2DMake(51.5074, -0.1278);
    if (!CLLocationCoordinate2DIsValid(source)
        || !CLLocationCoordinate2DIsValid(destination)) {
        return -1.0;
    }
    CLLocation *sourceLocation = [[CLLocation alloc]
        initWithLatitude:source.latitude
               longitude:source.longitude];
    CLLocation *destinationLocation = [[CLLocation alloc]
        initWithLatitude:destination.latitude
               longitude:destination.longitude];
    return [sourceLocation distanceFromLocation:destinationLocation];
}

File: full/corelocation/tests/CoreLocationMixedConsumer.swift
import CoreLocation
import Foundation

@_silgen_name("OpenCoreLocationMixedDistance")
private func openCoreLocationMixedDistance() -> Double

@main
struct CoreLocationMixedConsumer {
  static func main() {
    let swiftDistance = CLLocation(latitude: 40.7128, longitude: -74.0060)
      .distance(from: CLLocation(latitude: 51.5074, longitude: -0.1278))
    let objcDistance = openCoreLocationMixedDistance()
    precondition(swiftDistance > 0)
    precondition(abs(swiftDistance - objcDistance) < 0.001)
    print("CORELOCATION_MIXED_ABI_OK c=coordinate,constants,objc swift=module,class identity=one-dylib")
  }
}
